src/overlays/actors/ovl_En_In/z_en_in.c

```diff
@@ -21,92 +44,1623 @@ const ActorInit En_In_InitVars = {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    { 18, 64, 0, { 0, 0, 0 } },
+};
+
+static ColliderJntSphElementInit sJntSphElementsInit[1] = {
+    {
+        {
+            ELEMTYPE_UNK0,
+            { 0xF7CFFFFF, 0x00, 0x00 },
+            { 0x00000000, 0x00, 0x00 },
+            TOUCH_ON | TOUCH_SFX_NORMAL,
+            BUMP_NONE,
+            OCELEM_NONE,
+        },
+        { 12, { { 1600, 0, 0 }, 5 }, 200 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    {
+        COLTYPE_NONE,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_NONE,
+        OC1_NONE,
+        OC2_TYPE_2,
+        COLSHAPE_JNTSPH,
+    },
+    1,
+    sJntSphElementsInit,
+};
+
+CollisionCheckInfoInit2 sColChkInfoInit2 = {
+    0, 0, 0, 0, MASS_HEAVY,
+};
+
+static DamageTable sDamageTable = {
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+};
+
+static ActorAnimationEntryS sAnimations[] = {
+    { &D_06001D10, 1.0f, 0, -1, 0, 0 },  { &D_06001D10, 1.0f, 0, -1, 0, -4 }, { &D_06014F8C, 1.0f, 0, -1, 0, 0 },
+    { &D_06014F8C, 1.0f, 0, -1, 0, -4 }, { &D_06000CB0, 1.0f, 0, -1, 0, -4 }, { &D_060003B4, 1.0f, 0, -1, 0, -4 },
+    { &D_06001BE0, 1.0f, 0, -1, 0, -4 }, { &D_06015918, 1.0f, 0, -1, 0, -4 }, { &D_0601C0B0, 1.0f, 0, -1, 0, 0 },
+    { &D_0601C0B0, 1.0f, 0, -1, 0, -4 }, { &D_0601A140, 1.0f, 0, -1, 0, 0 },  { &D_0601A140, 1.0f, 0, -1, 0, -4 },
+    { &D_0601B904, 1.0f, 0, -1, 0, 0 },  { &D_0601B904, 1.0f, 0, -1, 0, -4 }, { &D_0601B3C4, 1.0f, 0, -1, 0, 0 },
+    { &D_0601B3C4, 0.0f, 0, -1, 2, 0 },  { &D_0601B3C4, 1.0f, 0, -1, 0, -4 }, { &D_06019EB4, 1.0f, 0, -1, 2, -4 },
+};
+
+static UNK_TYPE D_808F6C0C[] = {
+    0x0FA00004, 0x00010003, 0x17700004, 0x00010006, 0x0FA00004, 0x00010003, 0x17700004, 0x00010006,
+};
+
+s32 func_808F30B0(SkelAnime* skelAnime, s16 animIndex) {
+    s16 frameCount;
+    s32 ret = false;
+
+    if (animIndex >= 0 && animIndex < 18) {
+        ret = true;
+        frameCount = sAnimations[animIndex].frameCount;
+        if (frameCount < 0) {
+            frameCount = SkelAnime_GetFrameCount(&sAnimations[animIndex].animationSeg->common);
+        }
+        SkelAnime_ChangeAnim(skelAnime, sAnimations[animIndex].animationSeg, sAnimations[animIndex].playbackSpeed,
+                             sAnimations[animIndex].frame, frameCount, sAnimations[animIndex].mode,
+                             sAnimations[animIndex].transitionRate);
+    }
+    return ret;
+}
+
+s32 func_808F3178(EnIn* this, GlobalContext* globalCtx) {
+    u8 prevUnk260 = this->unk260;
+    u8 prevUnk261 = this->unk261;
+    u8 tmp;
+
+    tmp = func_8013DB90(globalCtx, &this->unk248, -6.0f);
+    this->unk260 = tmp;
+    if (this->unk260 != 0 && prevUnk260 == 0 && tmp & 0xFF) {
```

Comment:
Is this temp needed? Could you try assigning it directly?

Reply:
tmp is needed, otherwise regalloc.
![image](https://user-images.githubusercontent.com/31974442/121295641-3e478b80-c8a4-11eb-8f4b-21c692a4fdc3.png)